src/libs/conduit/conduit_node.cpp

```diff
@@ -13501,6 +13502,146 @@ Node::find_first_data_ptr() const
     return res;
 }
 
+//---------------------------------------------------------------------------//
+bool
+Node::equals(const Node &n) const
+{
+    Node res;
+    return compare(n, res);
+}
+
+//---------------------------------------------------------------------------//
+bool
+Node::compare(const Node &n, Node &info) const
+{
+    // TODO(JRC): Improve all of the messages for all of the different
+    // scenarios in this function.
+
+    bool res = true;
+    info.reset();
+
+    index_t m_dtid  = dtype().id();
+    index_t n_dtid  = n.dtype().id();
+
+    if(m_dtid != n_dtid)
+    {
+        info.set("Data type mismatch.");
+        res = false;
+    }
+    else if(m_dtid == DataType::OBJECT_ID)
+    {
+        NodeConstIterator child_itr;
+
+        child_itr = children();
+        while(child_itr.has_next())
+        {
+            const conduit::Node &m_child = child_itr.next();
+            const std::string child_path = child_itr.name();
+
+            Node &info_child = info[child_path];
+            if(!n.has_child(child_path))
+            {
+                info_child.set("Missing from other.");
+                res = false;
+            }
+            else
+            {
+                if(m_child.compare(n.fetch(child_path), info_child))
+                {
+                    info.remove(child_path);
+                }
+                else
+                {
+                    res = false;
+                }
+            }
+        }
+
+        child_itr = n.children();
+        while(child_itr.has_next())
+        {
+            const conduit::Node &n_child = child_itr.next();
+            const std::string child_path = child_itr.name();
+
+            Node &info_child = info[child_path];
+            if(!has_child(child_path))
+            {
+                info_child.set("Missing from self.");
+                res = false;
+            }
+            else
+            {
+                if(fetch(child_path).compare(n_child, info_child))
+                {
+                    info.remove(child_path);
+                }
+                else
+                {
+                    res = false;
+                }
+            }
+        }
+    }
+    else if(m_dtid == DataType::LIST_ID)
+    {
+        // NOTE: This case is a bit strange in the comparison; fortunately, it
+        // can be distilled to two primary cases:
+        // - Nodes Have 0 Diffs: The info node is completely empty.
+        // - Nodes Have >1 Diffs: The info node is a list where the differing
+        //   nodes between the comparison parents are filled in the list.
+        //   All other nodes in the list are empty.
+
+        index_t m_nchild = number_of_children();
+        index_t n_nchild = n.number_of_children();
+
+        size_t i = 0;
+        for(; i < (size_t)std::min(m_nchild, n_nchild); i++)
+        {
+            const Node &m_child = child(i);
```

Comment:
Small request: I usually limit  using the `m_` prefix to member vars, not local ones.  Would it be ok to change these?

Reply:
Point well taken.  Also, in retrospect, I don't think it's a great idea to differentiate variables using the `m` and `n` characters, so I'll definitely update this.